src/java/org/apache/cassandra/service/accord/AccordConfigurationService.java

```diff
@@ -67,23 +66,24 @@
 @Simulate(with=MONITORS)
 public class AccordConfigurationService extends AbstractConfigurationService<AccordConfigurationService.EpochState, AccordConfigurationService.EpochHistory> implements ChangeListener, AccordEndpointMapper, AccordSyncPropagator.Listener, Shutdownable
 {
-    private static final Logger logger = LoggerFactory.getLogger(AccordConfigurationService.class);
-
     private final AccordSyncPropagator syncPropagator;
     private final DiskStateManager diskStateManager;
 
+    @GuardedBy("this")
     private EpochDiskState diskState = EpochDiskState.EMPTY;
 
     private enum State { INITIALIZED, LOADING, STARTED, SHUTDOWN }
 
+    @GuardedBy("this")
     private State state = State.INITIALIZED;
     private volatile EndpointMapping mapping = EndpointMapping.EMPTY;
 
     public enum SyncStatus { NOT_STARTED, NOTIFYING, COMPLETED }
 
     static class EpochState extends AbstractConfigurationService.AbstractEpochState
     {
-        SyncStatus syncStatus = SyncStatus.NOT_STARTED;
+        @GuardedBy("AccordConfigurationService.this")
+        private SyncStatus syncStatus = SyncStatus.NOT_STARTED;
         protected final AsyncResult.Settable<Void> localSyncNotified = AsyncResults.settable();
```

Comment:
is it safe that we set this (and notify its callbacks) while holding the `AccordConfigurationService` lock?

Reply:
It looks like today this is only used in test methods, but I would prefer that we rename methods that can register callbacks as unsafe, and document why they are potentially unsafe, as it could cause surprises later.

Reply:
renamed the public api to `unsafeLocalSyncNotified` and added a javadoc saying why it is unsafe